src/query/storage/m3/multi_fetch_result.go

```diff
@@ -104,47 +107,52 @@ func (r *multiResult) FinalResultWithAttrs() (
 		}
 	}
 
-	return iters, attrs, nil
+	return result, attrs, nil
 }
 
-func (r *multiResult) FinalResult() (encoding.SeriesIterators, error) {
+func (r *multiResult) FinalResult() (SeriesFetchResult, error) {
 	r.Lock()
 	defer r.Unlock()
 
+	result := SeriesFetchResult{Metadata: r.metadata}
 	err := r.err.LastError()
 	if err != nil {
-		return nil, err
+		return result, err
 	}
-	if r.finalResult != nil {
-		return r.finalResult, nil
+
+	if r.mergedIterators != nil {
+		result.SeriesIterators = r.mergedIterators
+		return result, nil
 	}
 
 	if len(r.seenIters) == 0 {
-		return encoding.EmptySeriesIterators, nil
+		return result, nil
```

Comment:
Does `result.SeriesIterators = encoding.EmptySeriesIterators` need to be set here before returning?

Reply:
Oops, nice catch